-Wsign-compare: warning: comparison of integer expressions of different signedness

Hello,

With the default set of compiler flags on Debian, the following warnings appear:

```
../src/cff_charstring.cc: In function ‘bool {anonymous}::ExecuteCharStringOperator(ots::OpenTypeCFF&, int32_t, size_t, const ots::CFFIndex&, const ots::CFFIndex&, ots::Buffer*, ots::Buffer*, std::stack<int>*, bool*, bool*, size_t*, bool*, bool*, int32_t*, bool)’:
../src/cff_charstring.cc:376:31: warning: comparison of integer expressions of different signedness: ‘__gnu_cxx::__alloc_traits<std::allocator<int>, int>::value_type’ {aka ‘int’} and ‘std::vector<short unsigned int>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
  376 |     if (argument_stack->top() >= cff.region_index_count.size()) {
      |         ~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/cff_charstring.cc:393:25: warning: comparison of integer expressions of different signedness: ‘int32_t’ {aka ‘int’} and ‘std::vector<short unsigned int>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]
  393 |     if (*in_out_vsindex >= cff.region_index_count.size()) {
      |         ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../src/cff_charstring.cc:398:20: warning: comparison of integer expressions of different signedness: ‘const size_t’ {aka ‘const long unsigned int’} and ‘int’ [-Wsign-compare]
  398 |     if (stack_size < n * (k + 1) + 1) {
      |         ~~~~~~~~~~~^~~~~~~~~~~~~~~~~
../src/cff.cc: In function ‘bool {anonymous}::ParsePrivateDictData(ots::Buffer&, size_t, size_t, {anonymous}::DICT_DATA_TYPE, ots::OpenTypeCFF*)’:
../src/cff.cc:548:29: warning: comparison of integer expressions of different signedness: ‘std::vector<std::pair<unsigned int, {anonymous}::DICT_OPERAND_TYPE> >::size_type’ {aka ‘long unsigned int’} and ‘int’ [-Wsign-compare]
  548 |         if (operands.size() < n * (k + 1) + 1) {
      |             ~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
```

No other warnings are found, so it is quite a good news! As this code is very specific, I am not sure about what the solution would be between using int64_t everywhere, or doing something smarter.

Best regards.

--- a/src/cff.cc
+++ b/src/cff.cc
@@ -624,7 +624,7 @@ bool ParsePrivateDictData(
           return OTS_FAILURE();
         }
         uint16_t n = static_cast<uint16_t>(operands.back().as_integer());
-        if (operands.size() < n * (k + 1) + 1) {
+        if (operands.size() < n * (k + 1u) + 1u) {
           return OTS_FAILURE();
         }
         size_t operands_size = operands.size();

